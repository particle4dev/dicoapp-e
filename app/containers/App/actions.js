/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 * https://github.com/redux-utilities/flux-standard-action
 */

import {
  LOGIN,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  LOGOUT,
  LOAD_BALANCE,
  LOAD_BALANCE_SUCCESS,
  LOAD_BALANCE_ERROR,
  LOAD_COIN_BALANCE_SUCCESS,
  LOAD_WITHDRAW,
  LOAD_WITHDRAW_SUCCESS,
  LOAD_WITHDRAW_ERROR
} from './constants';

/**
 * Load the repositories, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_REPOS
 */
export function login(passphrase) {
  return {
    type: LOGIN,
    payload: {
      passphrase
    }
  };
}

/**
 * Dispatched when the repositories are loaded by the request saga
 *
 * @param  {array} repos The repository data
 * @param  {string} username The current username
 *
 * @return {object}      An action object with a type of LOGIN_SUCCESS passing the repos
 */
export function loginSuccess(user) {
  return {
    type: LOGIN_SUCCESS,
    payload: {
      user
    }
  };
}

/**
 * Dispatched when loading the repositories fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOGIN_ERROR passing the error
 */
export function loginError(error) {
  return {
    type: LOGIN_ERROR,
    error
  };
}

/**
 * Logout, this action starts the request saga
 *
 * @return {object} An action object with a type of LOGOUT
 */
export function logout() {
  return {
    type: LOGOUT
  };
}

/**
 *
 */

export function loadBalance() {
  return {
    type: LOAD_BALANCE
  };
}

export function loadBalanceSuccess() {
  return {
    type: LOAD_BALANCE_SUCCESS
  };
}

export function loadCoinBalanceSuccess(payload: {
  address: string,
  balance: number,
  coin: string
}) {
  return {
    type: LOAD_COIN_BALANCE_SUCCESS,
    payload
  };
}

export function loadBalanceError(message: string) {
  return {
    type: LOAD_BALANCE_ERROR,
    error: {
      message
    }
  };
}

export function loadWithdraw(payload: {
  amount: number,
  address: string,
  coin: string
}) {
  return {
    type: LOAD_WITHDRAW,
    payload
  };
}

export function loadWithdrawSuccess(payload: {
  amount: number,
  address: string,
  coin: string
}) {
  return {
    type: LOAD_WITHDRAW_SUCCESS,
    payload
  };
}

export function loadWithdrawError(message: string) {
  return {
    type: LOAD_WITHDRAW_ERROR,
    error: {
      message
    }
  };
}
