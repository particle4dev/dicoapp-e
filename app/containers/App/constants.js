/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */
export const APP_STATE_NAME = 'global';

export const LOGIN = 'dicoapp/App/LOGIN';
export const LOGIN_SUCCESS = 'dicoapp/App/LOGIN_SUCCESS';
export const LOGIN_ERROR = 'dicoapp/App/LOGIN_ERROR';

export const LOGOUT = 'dicoapp/App/LOGOUT';

export const LOAD_BALANCE = 'dicoapp/App/LOAD_BALANCE';
export const LOAD_BALANCE_SUCCESS = 'dicoapp/App/LOAD_BALANCE_SUCCESS';
export const LOAD_COIN_BALANCE_SUCCESS =
  'dicoapp/App/LOAD_COIN_BALANCE_SUCCESS';
export const LOAD_BALANCE_ERROR = 'dicoapp/App/LOAD_BALANCE_ERROR';

export const LOAD_WITHDRAW = 'dicoapp/App/LOAD_WITHDRAW';
export const LOAD_WITHDRAW_SUCCESS = 'dicoapp/App/LOAD_WITHDRAW_SUCCESS';
export const LOAD_WITHDRAW_ERROR = 'dicoapp/App/LOAD_WITHDRAW_ERROR';

export const LOAD_SWAP_SUCCESS = 'dicoapp/App/LOAD_SWAP_SUCCESS';
