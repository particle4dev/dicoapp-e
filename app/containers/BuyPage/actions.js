// @flow
import type { List } from 'immutable';

import {
  LOAD_PRICE,
  LOAD_PRICES,
  LOAD_COIN_SYMBOL,
  LOAD_BEST_PRICE,
  LOAD_PRICES_SUCCESS,
  LOAD_PRICES_ERROR
} from './constants';

export function loadPrice(coin: string) {
  return {
    type: LOAD_PRICE,
    payload: {
      coin
    }
  };
}

export function loadPrices() {
  return {
    type: LOAD_PRICES
  };
}

export function loadCoinSymbol(coins: List<*>) {
  return {
    type: LOAD_COIN_SYMBOL,
    payload: {
      coins
    }
  };
}

export function loadBestPrice(coin: string, name: string, bestPrice: number) {
  return {
    type: LOAD_BEST_PRICE,
    payload: {
      coin,
      name,
      bestPrice
    }
  };
}

export function loadPricesSuccess() {
  return {
    type: LOAD_PRICES_SUCCESS,
    payload: {}
  };
}

export function loadPricesError(message: string) {
  return {
    type: LOAD_PRICES_ERROR,
    error: {
      message
    }
  };
}
