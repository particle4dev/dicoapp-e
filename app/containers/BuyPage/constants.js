// eslint-disable-next-line import/prefer-default-export

export const APP_STATE_NAME = 'buy';
export const LOAD_PRICES = 'dicoapp/BuyPage/LOAD_PRICES';
export const LOAD_COIN_SYMBOL = 'dicoapp/BuyPage/LOAD_COIN_SYMBOL';
export const LOAD_PRICES_SUCCESS = 'dicoapp/BuyPage/LOAD_PRICES_SUCCESS';
export const LOAD_PRICES_ERROR = 'dicoapp/BuyPage/LOAD_PRICES_ERROR';
export const LOAD_PRICE = 'dicoapp/BuyPage/LOAD_PRICE';
export const LOAD_BEST_PRICE = 'dicoapp/BuyPage/LOAD_BEST_PRICE';
