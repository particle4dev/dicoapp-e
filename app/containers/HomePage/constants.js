// eslint-disable-next-line import/prefer-default-export
export const CHANGE_USERNAME = 'dicoapp/Home/CHANGE_USERNAME';
