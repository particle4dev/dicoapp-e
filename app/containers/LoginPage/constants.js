// eslint-disable-next-line import/prefer-default-export
export const APP_STATE_NAME = 'login';
