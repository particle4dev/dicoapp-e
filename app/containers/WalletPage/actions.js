import {
  LOAD_TRANSACTIONS,
  LOAD_TRANSACTIONS_SUCCESS,
  LOAD_TRANSACTIONS_ERROR
} from './constants';

export function loadTransactions() {
  return {
    type: LOAD_TRANSACTIONS
  };
}

export function loadTransactionsSuccess(transactions) {
  return {
    type: LOAD_TRANSACTIONS_SUCCESS,
    payload: {
      transactions
    }
  };
}

export function loadTransactionsError(message) {
  return {
    type: LOAD_TRANSACTIONS_ERROR,
    error: {
      message
    }
  };
}
