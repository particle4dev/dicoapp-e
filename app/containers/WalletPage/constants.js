// eslint-disable-next-line import/prefer-default-export
export const APP_STATE_NAME = 'wallet';
export const LOAD_WALLET = 'dicoapp/WalletPage/LOAD_WALLET';
export const LOAD_WALLET_SUCCESS = 'dicoapp/WalletPage/LOAD_WALLET_SUCCESS';
export const LOAD_WALLET_ERROR = 'dicoapp/WalletPage/LOAD_WALLET_ERROR';
