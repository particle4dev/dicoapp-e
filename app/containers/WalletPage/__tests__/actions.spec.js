import { loadTransactions } from '../actions';
import { LOAD_TRANSACTIONS } from '../constants';

describe('WalletPage Actions', () => {
  describe('loadTransactions', () => {
    it('should loadTransactions should create loadTransactions action', () => {
      expect(loadTransactions()).toMatchSnapshot();
    });

    it('should return the correct type and the passed name', () => {
      const expectedResult = {
        type: LOAD_TRANSACTIONS
      };

      expect(loadTransactions()).toEqual(expectedResult);
    });
  });
});
