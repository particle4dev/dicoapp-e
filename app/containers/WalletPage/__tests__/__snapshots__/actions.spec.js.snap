// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletPage Actions loadTransactions should loadTransactions should create loadTransactions action 1`] = `
Object {
  "type": "dicoapp/WalletPage/LOAD_TRANSACTIONS",
}
`;
